File: src/minigames/musical_minecart/java/work/lclpnet/ap2/game/musical_minecart/cmd/SkipSongCommand.java
package work.lclpnet.ap2.game.musical_minecart.cmd;

import com.mojang.brigadier.context.CommandContext;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.network.chat.Component;
import work.lclpnet.kibu.cmd.type.CommandRegistrar;
import work.lclpnet.kibu.cmd.type.KibuCommand;

import static net.minecraft.commands.Commands.literal;

public class SkipSongCommand implements KibuCommand {

    private final Runnable skipCurrent;

    public SkipSongCommand(Runnable skipCurrent) {
        this.skipCurrent = skipCurrent;
    }

    @Override
    public void register(CommandRegistrar registrar) {
        registrar.registerCommand(literal("ap2:skip_song")
                .requires(s -> s.hasPermission(2))
                .executes(this::skip));
    }

    private int skip(CommandContext<CommandSourceStack> ctx) {
        ctx.getSource().sendSystemMessage(Component.literal("Skipped the current song."));
        skipCurrent.run();
        return 1;
    }
}

File: src/minigames/musical_minecart/src/main/kotlin/work/lclpnet/ap2/game/musical_minecart/cmd/SkipSongCommand.kt
package work.lclpnet.ap2.game.musical_minecart.cmd

import net.minecraft.commands.Commands
import net.minecraft.network.chat.Component
import work.lclpnet.kibu.cmd.type.CommandRegistrar
import work.lclpnet.kibu.cmd.type.KibuCommand

class SkipSongCommand(val skipCurrent: Runnable) : KibuCommand {

    override fun register(registrar: CommandRegistrar) {
        registrar.registerCommand(
            Commands.literal("ap2:skip_song")
                .requires(Commands.hasPermission(Commands.LEVEL_GAMEMASTERS))
                .executes { ctx ->
                    ctx.source.sendSystemMessage(Component.literal("Skipped the current song."))
                    skipCurrent.run()
                    1
                })
    }
}
